web/elm/src/Build/Build.elm

```diff
@@ -462,23 +462,32 @@ update msg ( model, effects ) =
                 (Build.Output.Output.handleStepTreeMsg <| StepTree.updateTooltip newModel)
                 ( newModel, effects )
 
-        TriggerBuild job ->
-            case job of
+        Click TriggerBuildButton ->
+            case currentJob model of
                 Nothing ->
                     ( model, effects )
 
                 Just someJob ->
                     ( model, effects ++ [ DoTriggerBuild someJob ] )
 
-        AbortBuild buildId ->
-            ( model, effects ++ [ DoAbortBuild buildId ] )
+        Click AbortBuildButton ->
+            case
+                model.currentBuild
+                    |> RemoteData.toMaybe
+                    |> Maybe.map (.build >> .id)
+            of
+                Just buildId ->
```

Comment:
`|> Maybe.map (\buildId -> [ DoAbortBuild buildId ])
|> Maybe.withDefault []`

would also work. Not sure if you think it's cleaner 🤷‍♂️

Reply:
cae6a74402426c7c93898d567fecea4c1a6ae32a

Reply:
LGTM!